# Обработка ошибок

Если при выполнении скрипта в одной из фукнций возникнет ошибка, выполнение всего скрипта прервется и вернется ошибка. 
Для того, чтобы обработать ошибку и продолжить выполенние скрипта, 
служат конструкции try-catch-finally и упрощенный вариант error(),
которые будут рассмотрены далее



## try-catch-finally?

<<<<<<< HEAD
Оператор try определяет блок кода, который во время выполнения будет проверяться на возникновение ошибок.

Оператор catch определяет блок кода, который будет выполняться, если в блоке оператора try возникнет ошибка.

Операторы try и catch всегда используются в паре:

```javaScript
try {
     // проверяемый блок кода
}
 catch(err) {
     // блок кода для обработки ошибок
} 
```
Через параметр err можно получить информацию о возникшей ошибке:

err.message - сообщение об ошибке
err.lineNumber - номер строки
err.fileName - имя файла


Оператор finally позволяет выполнить код после операторов try и catch, независимо от результатов операторов try / catch.



## Error object?


=======
```js
try {
    // код
} catch(error) {
    // обработка ошибки
}
```

Выполняется блок `try`, если в нем нет ошибок блок `catch` игнорируется и выполнение скрипта продолжается. В противном случае выполнение блока `try` прекращается и начинает выполнятся блок `catch`, при этом в переменную `error` передается объект ошибки для анализа.

## Error object?
>>>>>>> 4508ae86687623c7d2131aa3e63f4f0b5c2484a2
